caas/kubernetes/provider/constraints.go

```diff
@@ -23,5 +27,81 @@ var unsupportedConstraints = []string{
 func (k *kubernetesClient) ConstraintsValidator(ctx envcontext.ProviderCallContext) (constraints.Validator, error) {
 	validator := constraints.NewValidator()
 	validator.RegisterUnsupported(unsupportedConstraints)
-	return validator, nil
+	return &constraintsValidator{Validator: validator}, nil
+}
+
+type constraintsValidator struct {
+	constraints.Validator
+}
+
+// Validate returns an error if the given constraints are not valid, and also
+// any unsupported attributes.
+func (v *constraintsValidator) Validate(cons constraints.Value) ([]string, error) {
+	if cons.Tags != nil {
+
+		var validateTopologyKeyTagFunc = func(key string, topologySpreadKey string, array []string) error {
+			var topologySpreadValue, otherValue *string = nil, nil
+			// Search for the topologySpreadValue for topologyKeyTag
+			for tag := range array {
+				if strings.HasPrefix(array[tag], topologySpreadKey) {
```

Comment:
This is fragile due to white space. 
We have an upstream keyvalues package in juju/utils, Or just implement a small helper which does something similar. We want to split on `=` and then trim white space.
Same with all the kv parsing below.